numba/targets/arraymath.py

```diff
@@ -1427,6 +1427,40 @@ def _prepare_cov_input(m, y, rowvar, dtype, ddof, _DDOF_HANDLER, _M_DIM_HANDLER)
     _DDOF_HANDLER(ddof)
     return _prepare_cov_input_inner(m, y, rowvar, dtype)
 
+def scalar_result_expected(mandatory_input, optional_input):
+    if isinstance(mandatory_input, types.Array) and mandatory_input.ndim == 1:
+        if optional_input in (None, types.none):
```

Comment:
Can this just be hoisted out all these functions as a bool and reused?:
```opt_is_none = optional_input in (None, types.none)```

Reply:
Done.